lib/api/BackbeatAPI.js

```diff
@@ -1088,22 +1124,27 @@ class BackbeatAPI {
      */
     pauseService(details, body, cb) {
         const service = details.service;
-        let sites;
-        if (details.site === 'all') {
-            sites = details.extensions[service].filter(s => s !== 'all');
-        } else {
-            sites = [details.site];
-        }
-        sites.forEach(site => {
+        const sites = this.getRequestedSites(details);
+        async.eachLimit(sites, 10, (site, next) => {
+            if (service === 'lifecycle') {
```

Comment:
this does not look really nice, with this state management intermingled everywhere (which was indeed already the case)

I am wondering if it would not be better to introduce some kind of interface "LocationStatusManager", called by this api : so we could have 2 implementations (redis/zookeeper and mongo), and easily switch one or the other...

That would be much cleaner IMO, but i am wondering if this is worth it (and how long that would take), esp. since we should eventually change this for CRR and ingestion as well... On the positiive side, it would make it easier to keep both codes around, progressively migrate (e.g. just ingestion) and possibly keep the code compatible with RING (where we will not have mongo, in case we want to upgrade RING to backbeat 8.x)